doc/build/kconfig/setting.rst

```diff
@@ -259,9 +259,24 @@ symbol properties, so the above ``default`` is equivalent to
       	...
       	depends on DEP2
 
+   Symbols without any listed dependencies are treated as having a dependency
```

Comment:
what is wrong with the `warning` above ?
![image](https://user-images.githubusercontent.com/9802583/195338442-37a3d33a-9b93-483d-a020-79c192f8221c.png)

Reply:
maybe you should look at the definition of `depends on`:
> Dependencies are applied to all other options within this menu entry (which also accept an “if” expression)

Ref: https://www.kernel.org/doc/html/latest/kbuild/kconfig-language.html

which basically means that the new example you are creating:
```
config FOO
    ...
    depends on DEP1
```
is identical to:
```
config FOO
    ... if DEP1
```
which suddenly makes it very clear why Kconfig behave the way it does, cause now you have:
```
config FOO
    ... if DEP1
```
and:
```
 config FOO
      	...
      	default y if BAR
```

Reply:
The warning above makes no mention of what the behaviour is when an entry doesn't have a `depends on`. Regardless of what the resolution is here, there should be some mention of this (as proven by @carlocaione running into the same behaviour and considering it unintuitive).

Your example doesn't really clear it up for me, why should the concrete example of
```
config CACHE_MANAGEMENT
    bool "Cache management features"
    depends on DCACHE || ICACHE

config CACHE_MANAGEMENT
    default y
```
Be equivalent to:
```
config CACHE_MANAGEMENT
    bool "Cache management features"
    default y
```

Also as far as I can tell the linked reference makes no mention of how multiple symbols are handled, beyond:
```
A config option can be defined multiple times with the same name, but every definition can have only a single input prompt and the type must not conflict.
```

Reply:
How about the whole part and example, https://www.kernel.org/doc/html/latest/kbuild/kconfig-language.html:
(Unfortunately I cannot link directly to the description)
> Dependencies are applied to all other options within this menu entry (which also accept an “if” expression), so these two examples are equivalent:
>
> ```
> bool "foo" if BAR
> default y if BAR
> ```
> and:
> ```
> depends on BAR
> bool "foo"
> default y
> ```

So when you write:
```
config CACHE_MANAGEMENT
    bool "Cache management features"
    depends on DCACHE || ICACHE

config CACHE_MANAGEMENT
    default y
``` 

you are actually writing:
```
config CACHE_MANAGEMENT
    bool "Cache management features" if DCACHE || ICACHE

config CACHE_MANAGEMENT
    default y
```

guess it's clearer now that you are actually defining two `CACHE_MANAGEMENT` settings.
One which has a prompt if `DCACHE` or `ICACHE` is enabled.
And a promptless setting which simply defaults `CACHE_MANAGEMENT` unconditionally to `y`.